### ISSUE TITLE
Workloads installation broken after attempt to install maui from 'main' branch

### ISSUE THREAD

**Issue by Dreamescaper** [dotnet/maui#7008]
### Description

I had VS 17.2 Preview installed with MAUI RC 2, and it worked perfectly fine. Then I've tried to install Maui workloads from 'main' branch using steps from https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md . Unfortunately, installation failed with an error about unsigned packages or something like that. 
Now builds fail with messages similar to "To build this project, the following workloads must be installed: android/ios/maccatalyst/etc", "Platform version if not present for one or more target frameworks, even though the have specified a platform".

I've tried the following actions:
- Remove all workloads using "dotnet workload uninstall".
- Remove MAUI workload from VS install, then install it back.
- Repair the whole VS installation.
- Uninstall Visual Studio, uninstall all dotnet versions, remove all VS/nuget/dotnet related files from Program Files, Program Data, AppData folders, install Visual Studio back.

None of that helped, build still fails with the same issue. I have no idea what else could I remove to get clean VS/dotnet installation. Probably something in the registry? Not sure.

I noticed that my 'sdk-manifests' does not match same folder on another laptop (where everything works fine). So I've simply replaced this folder content manually. It allowed me to build projects, everything seemed to be fine. But probably it haven't fixed some underlying issue, because after today's VS 17.3 update I have the same issue again.
Probably I'll be able to workaround it by copying 'sdk-manifest' folder again, but I'd like to be able to fix my installation (hopefully without erasing my SSD drive).

**Comment by jfversluis** [comment]
@mattleibow you're usually good with this stuff...

I would think you have done a lot already, you might additionally try to run the [clean script](https://github.com/Redth/dotnet-maui-check/blob/main/Clean-Old-DotNet6-Previews.ps1) on the maui-check repo and/or check if there is something in there you missed.

**Comment by Dreamescaper** [comment]
Ok, so I have the same issue now, after VS 17.3 Preview 1.1 update. I have removed MAUI workload using VS installer, executed clean script, and installed MAUI workload back. It's still broken.

"/dotnet/sdk-manifests" folder is empty, same as "dotnet/metadata".
"dotnet workload list" returns no entries.

Same as before, I was able to workaround this issue by manually copying those folders from another laptop. But I suppose that I'll have the same issue after next update :(
